Failing google-analytics integration test (Chrome/Travis CI)

I've seen the [following failure](https://travis-ci.org/GoogleChrome/workbox/builds/567988253#L2679) at least a couple of times since we've moved to the OS X Travis CI environment. It happens in Chrome Stable:

```
  1) [workbox-google-analytics] initialize
2680       replay failed Google Analytics hits:
2681
2682      AssertionError: expected 'pixel' to equal 'beacon'
2683      + expected - actual
2684
2685      -pixel
2686      +beacon
2687      
2688      at Context.<anonymous> (test/workbox-google-analytics/integration/test-all.js:149:38)
2689      at <anonymous>
```

--- a/test/workbox-google-analytics/integration/test-all.js
+++ b/test/workbox-google-analytics/integration/test-all.js
@@ -96,16 +96,17 @@ describe(`[workbox-google-analytics] initialize`, function() {
       action: 'clear-spied-requests',
     });
 
-    // Check the "simulate offline" checkbox and make some requests.
-    await simulateOfflineEl.click();
+    // Switch the service worker into "offline" mode.
+    await webdriver.executeAsyncScript(messageSW, {
+      action: 'simulate-offline',
+      value: true,
+    });
 
     await webdriver.executeAsyncScript((done) => {
       window.gtag('event', 'beacon', {
         transport_type: 'beacon',
         event_label: Date.now(),
-        event_callback: () => {
-          setTimeout(done, 50);
-        },
+        event_callback: done,
       });
     });
 
@@ -113,9 +114,7 @@ describe(`[workbox-google-analytics] initialize`, function() {
       window.gtag('event', 'pixel', {
         transport_type: 'image',
         event_label: Date.now(),
-        event_callback: () => {
-          setTimeout(done, 50);
-        },
+        event_callback: done,
       });
     });
 
@@ -131,8 +130,13 @@ describe(`[workbox-google-analytics] initialize`, function() {
     });
     expect(requests).to.have.lengthOf(0);
 
-    // Uncheck the "simulate offline" checkbox and then trigger a sync.
-    await simulateOfflineEl.click();
+    // Switch the service worker into "online" mode.
+    await webdriver.executeAsyncScript(messageSW, {
+      action: 'simulate-offline',
+      value: false,
+    });
+
+    // Dispatch a sync event.
     await webdriver.executeAsyncScript(messageSW, {
       action: 'dispatch-sync-event',
     });
@@ -160,6 +164,8 @@ describe(`[workbox-google-analytics] initialize`, function() {
     // and ensure those values reflect the original order of the hits.
     expect(requests[0].params.qt > 0).to.be.true;
     expect(requests[1].params.qt > 0).to.be.true;
-    expect(requests[0].originalTime < requests[1].originalTime).to.be.true;
+    // This should be <= instead of less than, since a ms didn't necessarily
+    // pass in between the two events.
+    expect(requests[0].originalTime <= requests[1].originalTime).to.be.true;
   });
 });

--- a/test/workbox-google-analytics/integration/test-all.js
+++ b/test/workbox-google-analytics/integration/test-all.js
@@ -6,7 +6,6 @@
   https://opensource.org/licenses/MIT.
 */
 
-const {By} = require('selenium-webdriver');
 const expect = require('chai').expect;
 const qs = require('qs');
 
@@ -74,9 +73,6 @@ describe(`[workbox-google-analytics] initialize`, function() {
       return 'SyncManager' in window;
     });
     if (!browserSupportsSync) this.skip();
-
-    const simulateOfflineEl =
-        await webdriver.findElement(By.id('simulate-offline'));
 
     // Send a hit while online to ensure regular requests work.
     await webdriver.executeAsyncScript((done) => {

--- a/test/workbox-google-analytics/integration/test-all.js
+++ b/test/workbox-google-analytics/integration/test-all.js
@@ -6,13 +6,13 @@
   https://opensource.org/licenses/MIT.
 */
 
+const {By} = require('selenium-webdriver');
 const expect = require('chai').expect;
 const qs = require('qs');
-const {By} = require('selenium-webdriver');
+
+const {runUnitTests} = require('../../../infra/testing/webdriver/runUnitTests');
 const activateAndControlSW = require('../../../infra/testing/activate-and-control');
 const waitUntil = require('../../../infra/testing/wait-until');
-const {runUnitTests} = require('../../../infra/testing/webdriver/runUnitTests');
-
 
 // Store local references of these globals.
 const {webdriver, server} = global.__workbox;
@@ -98,20 +98,27 @@ describe(`[workbox-google-analytics] initialize`, function() {
 
     // Check the "simulate offline" checkbox and make some requests.
     await simulateOfflineEl.click();
+
     await webdriver.executeAsyncScript((done) => {
       window.gtag('event', 'beacon', {
         transport_type: 'beacon',
         event_label: Date.now(),
-        event_callback: () => done(),
+        event_callback: () => {
+          setTimeout(done, 50);
+        },
       });
     });
+
     await webdriver.executeAsyncScript((done) => {
       window.gtag('event', 'pixel', {
         transport_type: 'image',
         event_label: Date.now(),
-        event_callback: () => done(),
+        event_callback: () => {
+          setTimeout(done, 50);
+        },
       });
     });
+
     // This request should not match GA routes, so it shouldn't be replayed.
     await webdriver.executeAsyncScript((done) => {
       fetch('https://httpbin.org/get').then(() => done());
@@ -152,7 +159,7 @@ describe(`[workbox-google-analytics] initialize`, function() {
     // Ensure the hit's qt params were present and greater than 0,
     // and ensure those values reflect the original order of the hits.
     expect(requests[0].params.qt > 0).to.be.true;
-    expect(requests[0].params.qt > 0).to.be.true;
+    expect(requests[1].params.qt > 0).to.be.true;
     expect(requests[0].originalTime < requests[1].originalTime).to.be.true;
   });
 });

--- a/test/workbox-google-analytics/integration/test-all.js
+++ b/test/workbox-google-analytics/integration/test-all.js
@@ -103,6 +103,7 @@ describe(`[workbox-google-analytics] initialize`, function() {
         transport_type: 'beacon',
         event_label: Date.now(),
         event_callback: () => {
+          // See https://github.com/GoogleChrome/workbox/issues/2168
           setTimeout(done, 50);
         },
       });
@@ -113,6 +114,7 @@ describe(`[workbox-google-analytics] initialize`, function() {
         transport_type: 'image',
         event_label: Date.now(),
         event_callback: () => {
+          // See https://github.com/GoogleChrome/workbox/issues/2168
           setTimeout(done, 50);
         },
       });

--- a/test/workbox-google-analytics/integration/test-all.js
+++ b/test/workbox-google-analytics/integration/test-all.js
@@ -102,7 +102,9 @@ describe(`[workbox-google-analytics] initialize`, function() {
       window.gtag('event', 'beacon', {
         transport_type: 'beacon',
         event_label: Date.now(),
-        event_callback: done,
+        event_callback: () => {
+          setTimeout(done, 50);
+        },
       });
     });
 
@@ -110,7 +112,9 @@ describe(`[workbox-google-analytics] initialize`, function() {
       window.gtag('event', 'pixel', {
         transport_type: 'image',
         event_label: Date.now(),
-        event_callback: done,
+        event_callback: () => {
+          setTimeout(done, 50);
+        },
       });
     });
 
@@ -160,8 +164,6 @@ describe(`[workbox-google-analytics] initialize`, function() {
     // and ensure those values reflect the original order of the hits.
     expect(requests[0].params.qt > 0).to.be.true;
     expect(requests[1].params.qt > 0).to.be.true;
-    // This should be <= instead of less than, since a ms didn't necessarily
-    // pass in between the two events.
-    expect(requests[0].originalTime <= requests[1].originalTime).to.be.true;
+    expect(requests[0].originalTime < requests[1].originalTime).to.be.true;
   });
 });

